grafana-plugin/pkg/plugin/resources.go

```diff
@@ -2,36 +2,231 @@ package plugin
 
 import (
 	"bytes"
+	"encoding/json"
 	"fmt"
+	"github.com/grafana/grafana-plugin-sdk-go/backend"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/log"
+	"github.com/grafana/grafana-plugin-sdk-go/backend/resource/httpadapter"
+	"io"
 	"net/http"
+	"net/url"
+	"strconv"
 )
 
-// curl -X GET -H "Accept: application/json"  http://oncall:oncall@localhost:3000/api/plugins/grafana-oncall-app/resources/ping | jq
-func (a *App) handlePing(w http.ResponseWriter, req *http.Request) {
-	w.Header().Add("Content-Type", "application/json")
-	
-	msg := fmt.Sprintf(`{"message":"%s"}`, "ok")
-	if _, err := w.Write([]byte(msg)); err != nil {
+type XInstanceContextJSONData struct {
+	StackId      string `json:"stack_id,omitempty"`
+	OrgId        string `json:"org_id,omitempty"`
+	GrafanaToken string `json:"grafana_token"`
+}
+
+type XGrafanaContextJSONData struct {
+	ID          int    `json:"UserID"`
+	IsAnonymous bool   `json:"IsAnonymous"`
+	Name        string `json:"Name"`
+	Login       string `json:"Login"`
+	Email       string `json:"Email"`
+	Role        string `json:"Role"`
+}
+
+type UserIDJSONData struct {
+	ID int `json:"intField"`
+}
+
+func SetXInstanceContextHeader(settings OnCallPluginSettings, req *http.Request) error {
+	xInstanceContext := XInstanceContextJSONData{
+		StackId:      strconv.Itoa(settings.StackID),
+		OrgId:        strconv.Itoa(settings.OrgID),
+		GrafanaToken: settings.GrafanaToken,
+	}
+	xInstanceContextHeader, err := json.Marshal(xInstanceContext)
+	if err != nil {
+		return err
+	}
+	req.Header.Set("X-Instance-Context", string(xInstanceContextHeader))
+	return nil
+}
+
+func SetXGrafanaContextHeader(user *backend.User, userID int, req *http.Request) error {
+	var xGrafanaContext XGrafanaContextJSONData
+	if user == nil {
+		xGrafanaContext = XGrafanaContextJSONData{
+			IsAnonymous: true,
+		}
+	} else {
+		xGrafanaContext = XGrafanaContextJSONData{
+			ID:          userID,
+			IsAnonymous: false,
+			Name:        user.Name,
+			Login:       user.Login,
+			Email:       user.Email,
+			Role:        user.Role,
```

Comment:
Should/Can we get information about teams membership of the current user?

Reply:
We should be able to get it as another API call